test/unit_tests/braket/experimental/autoqasm/test_gate_definitions.py

```diff
@@ -142,3 +148,72 @@ def test_gates(gate, qubits, params, expected_qasm) -> None:
         gate(*qubits, *params)
 
     assert expected_qasm in program_conversion_context.make_program().to_ir()
+
+
+@pytest.mark.parametrize(
+    "gate,qubits,params,control,control_state,power,expected_qasm",
```

Comment:
Nit: I think parameterizations are overused. It makes it really hard to tell whether this is correct or what is being tested, because I can't see all the logic at once. Compare to

```
def test_gate_modifiers() -> None:
    """Tests quantum gate modifiers."""
    with aq.build_program() as program_conversion_context:
        x(1, control=0)

    assert (
        "ctrl @ x __qubits__[0], __qubits__[1];"
        in program_conversion_context.make_program().to_ir()
    )
```

I know there's a lot of combinations to check here and this is the easiest way to be thorough. Perhaps a good in-between would be to find logical subgroupings of the parameterizations, and test those separately e.g. `test_gate_modifiers_with_neg_controls`, `test_gate_modifier_basis_state_types`, etc, so that the information is easier to parse.

That also allows for fewer parameters for some tests. That makes it much easier to review and hold all the relevant data in mind at once. So, you could try grouping together all the tests have have third arg `[]` and second to last arg `None`, if that makes sense

Reply:
Very good suggestion, thank you.